src/main/drivers/compass_hmc5883l.h

```diff
@@ -17,7 +17,7 @@
 
 #pragma once
 
-#include "drivers/io.h"
```

Comment:
My vote is to use the directory prefix, possibly with exception of '.h' matching '.c' file.

We can also modify Makefile to restrict `-I` directories as much as practical ...